backend/satellite_tools/reposync.py

```diff
@@ -741,6 +741,10 @@ def import_updates(self, plug):
             elif notices_type == 'patches':
                 self.upload_patches(notices)
 
+            # some errata could get retracted and this needs to be reflected in the newest package cache
+            refresh_newest_package = rhnSQL.Procedure("rhn_channel.refresh_newest_package")
+            refresh_newest_package(self.channel['id'], 'backend.importPatches')
```

Comment:
We should do this only when patches were imported or changed.
We skip unchanged patches. So we could change `self.upload_patches(notices)` to return the number of imported patches. If this number is `0` we do not need to execute this task.

Reply:
Sounds good :+1: